crates/bevy_animation/src/lib.rs

```diff
@@ -479,174 +485,186 @@ impl AnimationPlayer {
     }
 }
 
-fn entity_from_path(
-    root: Entity,
-    path: &EntityPath,
-    children: &Query<&Children>,
-    names: &Query<&Name>,
-    path_cache: &mut Vec<Option<Entity>>,
-) -> Option<Entity> {
-    // PERF: finding the target entity can be optimised
-    let mut current_entity = root;
-    path_cache.resize(path.parts.len(), None);
-
-    let mut parts = path.parts.iter().enumerate();
-
-    // check the first name is the root node which we already have
-    let Some((_, root_name)) = parts.next() else {
-        return None;
-    };
-    if names.get(current_entity) != Ok(root_name) {
-        return None;
+pub fn update_animation_targets(
+    mut commands: Commands,
+    candidate_targets: Query<
+        (
+            Entity,
+            &Parent,
+            Option<&Children>,
+            &Name,
+            Option<&AnimationTarget>,
+        ),
+        (
+            Or<(With<Parent>, With<Children>)>,
+            Or<(Changed<Parent>, Changed<Children>, Changed<Name>)>,
+            Without<AnimationPlayer>,
```

Comment:
Will this run on a root-level parent when the last child is removed, or on a leaf-level child when it is removed from it's parent?

Reply:
No in both cases. I'll add the check for the leaf-level child. But actually this makes me realize: We don't need to fire when `Children` is changed, only when `Parent` is changed, I don't think.

Reply:
Should be fixed now.

Reply:
|> We don't need to fire when Children is changed, only when Parent is changed, I don't think.

That makes sense. In most cases parents+children would be double counting. We should have a test for last-child-removed to make sure everything still works.